### PR TITLE
test: Add `FlatESLint` tests with missing config files

### PR DESCRIPTION
**Pull request by mdjermanovic**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[ ] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-change-proposal.md))
[ ] Add autofix to a rule
[ ] Add a CLI option
[ ] Add something to the core
[x] Other, please explain:

While reviewing https://github.com/eslint/eslint/pull/17142, I noticed we didn't have tests where `eslint.config.js` is missing. In particular, a test that would confirm that `FlatESLint` lint methods should throw if a config file was not found.

#### What changes did you make? (Give an overview)

Added a few tests for `FlatESLint#lintText()` and `FlatESLint#lintFiles()`.

### CODE REVIEW

**Review by nzakas — COMMENTED** [review]
The tests look good. Just left a couple suggestions.

File: tests/lib/eslint/flat-eslint.js [hunk 1]
```diff
@@ -648,6 +648,37 @@ describe("FlatESLint", () => {
...
+            });
+            await eslint.lintText("var foo = 'bar';");
+        });
+
+        it("should throw if overrideConfigFile is path to a file that doesn't exist", async () => {
+            eslint = new FlatESLint({
+                cwd: getFixturePath(""),
+                overrideConfigFile: "does-not-exist.js"
+            });
+            await assert.rejects(() => eslint.lintText("var foo = 'bar';"));
```
[the remark this answers is not in the document]
Comment (BYK) [reply tests/lib/eslint/flat-eslint.js:679]: Maybe `{code: "ENOENT"}`? Looks more accurate to me. We can even add a checker function that asserts the code and uses a regex on the message property. (docs: https://nodejs.org/api/assert.html#assertrejectsasyncfn-error-message)

File: tests/lib/eslint/flat-eslint.js [hunk 2]
```diff
@@ -763,6 +794,37 @@ describe("FlatESLint", () => {
...
+            });
+            await eslint.lintFiles("fixtures/undef*.js");
+        });
+
+        it("should throw if overrideConfigFile is path to a file that doesn't exist", async () => {
+            eslint = new FlatESLint({
+                cwd: getFixturePath(),
+                overrideConfigFile: "does-not-exist.js"
+            });
+            await assert.rejects(() => eslint.lintFiles("undef*.js"));
```
[the remark this answers is not in the document]
Comment (BYK) [reply tests/lib/eslint/flat-eslint.js:825]: I concur. We should check the error code and message when viable as _any_ error can mask actual errors code (working or test).

File: tests/lib/eslint/flat-eslint.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (nzakas) [reply tests/lib/eslint/flat-eslint.js:679]: I think anything that checks the expected error message is fine. I just want to make sure we are getting the error we're expecting. I'm agnostic as to the approach.

[the remark this answers is not in the document]
Comment (mdjermanovic) [reply tests/lib/eslint/flat-eslint.js]: Added `{ code: "ENOENT" }`, thanks for the suggestion!

File: tests/lib/eslint/flat-eslint.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (mdjermanovic) [reply tests/lib/eslint/flat-eslint.js]: Added `{ code: "ENOENT" }`

**Review by nzakas — APPROVED** [review]
LGTM. Thanks

**Pull request merged**